### ISSUE TITLE
[BUG]FPDT training bug

### ISSUE THREAD

**Issue by wukong1992** [deepspeedai/DeepSpeed#7706]
**Describe the bug**
trianing with fpdt, error:"one of the variables needed for gradient computation has been modified by an inplace operation: [CUDABFloat16Type [40960, 1, 1280]], which is output 0 of FusedLayerNormAffineFunctionBackward"

model:
fpdt_attention
post attention layernorm
fpdt_ffn

error raised  in post attention layernorm, I suspect that the operations in fpdt_ffn might be causing the bug.

trace：
return FusedLayerNormAffineFunction.apply(input, weight, self.bias, self.normalized_shape, self.eps, self.mem_efficient_ln)
  File "/usr/local/lib/python3.12/dist-packages/torch/autograd/function.py", line 575, in apply
    return super().apply(*args, **kwargs)  # type: ignore[misc]
 (Triggered internally at /opt/pytorch/pytorch/torch/csrc/autograd/python_anomaly_mode.cpp:122.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
